pymc_extras/statespace/models/DFM.py

```diff
@@ -0,0 +1,828 @@
+        if self._exog:
+            names.append("beta")
+            if self.exog_innovations:
+                names.append("beta_sigma")
+
+        return names
+
+    @property
+    def param_info(self) -> dict[str, dict[str, Any]]:
+        info = {
+            "x0": {
+                "shape": (self.k_states,),
+                "constraints": None,
+            },
+            "P0": {
+                "shape": (self.k_states, self.k_states),
+                "constraints": "Positive Semi-definite",
+            },
+            "factor_loadings": {
+                "shape": (self.k_endog, self.k_factors),
+                "constraints": None,
+            },
+            "factor_ar": {
+                "shape": (self.k_factors, self.factor_order * self.k_factors),
+                "constraints": None,
+            },
+            "error_ar": {
+                "shape": (
+                    self.k_endog,
+                    self.error_order * self.k_endog if self.error_var else self.error_order,
+                ),
+                "constraints": None,
+            },
+            "error_sigma": {
+                "shape": (self.k_endog,) if self.error_cov_type == "diagonal" else (),
+                "constraints": "Positive",
+            },
+            "error_cov": {
+                "shape": (self.k_endog, self.k_endog),
+                "constraints": "Positive Semi-definite",
+            },
+            "sigma_obs": {
+                "shape": (self.k_endog,),
+                "constraints": "Positive",
+            },
+            "beta": {
+                "shape": (self.k_exog_states if self.k_exog is not None else 0,),
+                "constraints": None,
+            },
+            "beta_sigma": {
+                "shape": (self.k_exog_states if self.k_exog is not None else 0,),
+                "constraints": "Positive",
+            },
+        }
+
+        for name in self.param_names:
+            info[name]["dims"] = self.param_dims[name]
+
+        return {name: info[name] for name in self.param_names}
+
+    @property
+    def state_names(self) -> list[str]:
+        """
+        Returns the names of the hidden states: first factor states (with lags),
+        idiosyncratic error states (with lags), then exogenous states.
+        """
+        names = []
+        # Factor states
+        for i in range(self.factor_order):
+            for lag in range(max(self.factor_order, 1)):
+                names.append(f"L{lag}.factor_{i}")
+
+        # Idiosyncratic error states
+        if self.error_order > 0:
+            for i in range(self.k_endog):
+                for lag in range(self.error_order):
+                    names.append(f"L{lag}.error_{i}")
+
+        if self._exog:
+            if self.shared_exog_states:
+                # Shared exogenous states
+                for i in range(self.k_exog):
+                    names.append(
+                        f"exog_{i}.shared"
```

Comment:
```suggestion
                        f"exog_{i}[shared]"
```

We're using this square bracket notation for now